app/javascript/packages/password-strength/password-strength-element.ts

```diff
@@ -0,0 +1,188 @@
+import zxcvbn from 'zxcvbn';
+import { t } from '@18f/identity-i18n';
+import type { ZXCVBNResult, ZXCVBNScore } from 'zxcvbn';
+
+/**
+ * A mapping of zxcvbn scores to corresponding CSS BEM modifier classes.
+ */
+const PASSWORD_SCORE_CLASS_MODIFIERS: Record<ZXCVBNScore, string> = {
+  0: 'password-strength--very-weak',
+  1: 'password-strength--weak',
+  2: 'password-strength--average',
+  3: 'password-strength--good',
+  4: 'password-strength--great',
+};
+
+/**
+ * A mapping of zxcvbn scores to corresponding strength indicators.
+ */
+const PASSWORD_SCORE_STRENGTH: Record<ZXCVBNScore, string> = {
+  0: t('instructions.password.strength.i'),
```

Comment:
In ruby, we'd need to defer evaluation of the `t()` call, because it would get evaluated as the default locale (English) so loading the page in say, Spanish, would have the wrong values. Do we need to do that here?

Reply:
The way this works in JavaScript, it'll be guaranteed that the correct locale data is available anywhere within a script.

Specifically, the way it works is that the locale data script for the current locale is loaded before the main script, and (synchronous) scripts are [guaranteed to be loaded in order](https://html.spec.whatwg.org/multipage/scripting.html#:~:text=If%20neither%20attribute%20is%20present%2C%20then%20the%20script%20is%20fetched%20and%20evaluated%20immediately%2C%20blocking%20parsing%20until%20these%20are%20both%20complete.):

https://github.com/18F/identity-idp/blob/9091437457826cf6ceda4e6503d14f7281c2d4ea/lib/asset_sources.rb#L16-L23